api/types/microvm.proto

```diff
@@ -66,7 +66,9 @@ message MicroVMSpec {
 message Kernel {
   // Image is the container image to use.
   string image = 1;
-  // Cmdline is the kernel command line args.
+  // Cmdline is the additional kernel command line args. Each provider has its
+  // own recommended list, they will be used automatically. This field is for
+  // additional values.
```

Comment:
I'm wondering if we change away from a string to a key:value structure (map[string]string in go) where value can be a empty string. And then we construct the cmd line as a string after we we take the defaults for the provider and merge them with the additional ones supplied by the user.

We'd want some sort of precedence logic as well. For example, we default Firecracker to using a data source of no cloud with the Firecracker metadata service but perhaps they have their own metadata service so in the additional args they would supply `ds=nocloud-net;s=http://supercoolmetadata/latest/` and it would overwite the default for firecracker.